src/main/java/com/echostreams/pulsar/jms/client/PulsarConnectionFactory.java

```diff
@@ -179,12 +180,43 @@ private PulsarClient prepareClient(Authentication tlsOrAthenzAuth) throws Pulsar
     /*
      * Checking if already set by JNDI context, else take from config file else take default value
      */
-    private String getUpdatedServiceUrl(){
-        if(SERVICE_URL == null){
-            SERVICE_URL = (PulsarConfig.SERVICE_URL == null)?DEFAULT_BROKER_URL:PulsarConfig.SERVICE_URL;
+    private String getUpdatedServiceUrl() {
+        if (SERVICE_URL == null) {
+            SERVICE_URL = (PulsarConfig.SERVICE_URL == null) ? DEFAULT_BROKER_URL : PulsarConfig.SERVICE_URL;
         }
 
         return SERVICE_URL;
     }
 
+    /**
+     * Set the properties that will represent the instance in JNDI
+     *
+     * @param props The properties to use when building the new isntance.
+     * @return a new, unmodifiable, map containing any unused properties, or empty if none were.
+     */
+    @Override
+    protected Map<String, String> buildFromProperties(Map<String, String> props) {
+        String remoteURI = props.remove(DEFAULT_BROKER_PROP);
```

Comment:
Why do you remove DEFAULT_BROKER_PROP from the map?

Reply:
Actually SERVICE_URL  is used to connect to pulsar client so removing this key and assigning this value to SERVICE_URL. If we don't remove after use it will throw error as unused and terminate the app to assure we only use standard jndi props